### PR TITLE
cluster: fix issue with k1 signature verification

### PR DESCRIPTION
**Pull request by HananINouman**
Fix k1 signature verification by adding support for yellow paper recovery id. Edit `cluster` tests.

category: bug
ticket: https://github.com/ObolNetwork/charon/issues/1149

### CODE REVIEW

File: testutil/random.go [hunk 1]
```diff
@@ -522,6 +522,16 @@ func RandomBitList() bitfield.Bitlist {
 	return resp
 }
 
+// RandomSecp256k1Signature returns a random secp256k1 ECDSA signature with the last byte set to 0.
+func RandomSecp256k1Signature() []byte {
+	var resp [65]byte
+	_, _ = rand.Read(resp[:])
+
+	resp[64] = byte(0)
```
Comment (corverroos) [inline testutil/random.go:530]: suggest making this random of either 0,1,27 or 28.

**Review by corverroos — APPROVED** [review]
Awesome work, well done on your first charon PR! 🚀 💪

File: cluster/helpers.go [hunk 2]
```diff
@@ -47,6 +47,20 @@ func uuid(random io.Reader) string {
 
 // verifySig returns true if the signature matches the digest and address.
 func verifySig(expectedAddr []byte, digest []byte, sig []byte) (bool, error) {
+	if len(sig) != 65 {
```
Comment (dB2510) [inline cluster/helpers.go:50]: add a global const variable for "65" for ex: `const k1SigLen = 65`

**Review by dB2510 — APPROVED** [review]
🚀

**Pull request merged**